src/TestProblem/Hydro/CCSN/dtSolver_Lightbulb.cpp

```diff
@@ -0,0 +1,179 @@
+#           else
+            const double Emag = NULL_REAL;
+#           endif
+
+            const double Dens_Code = Dens;
+            const double Eint_Code = Hydro_Con2Eint( Dens, Momx, Momy, Momz, Engy, false, NULL_REAL, Emag );
+            const double Ye        = YeDens / Dens;
+
+
+//          compute the neutrino heating rate
+#           if ( NUC_TABLE_MODE == NUC_TABLE_MODE_TEMP )
+            const int  NTarget = 2;
+#           else
+            const int  NTarget = 3;
+#           endif
+                  int  In_Int[NTarget+1];
+                  real In_Flt[3], Out[NTarget+1];
+
+            In_Flt[0] = Dens_Code;
+            In_Flt[1] = Eint_Code;
+            In_Flt[2] = Ye;
+
+            In_Int[0] = NTarget;
+            In_Int[1] = NUC_VAR_IDX_XN;
+            In_Int[2] = NUC_VAR_IDX_XP;
+#           if ( NUC_TABLE_MODE == NUC_TABLE_MODE_ENGY )
+            In_Int[3] = NUC_VAR_IDX_EORT;
+#           endif
+
+            EoS_General_CPUPtr( NUC_MODE_ENGY, Out, In_Flt, In_Int, EoS_AuxArray_Flt, EoS_AuxArray_Int, h_EoS_Table );
+
+            const double Xn       = Out[0];               // neutron number fraction
+            const double Xp       = Out[1];               // proton  number fraction
+            const double Temp_MeV = Out[2] * Kelvin2MeV;  // temperature in MeV
+
+            const double rate_heating = 1.544e20 * ( SrcTerms.Lightbulb_Lnue / 1.0e52 ) * ( 1.0e14 / r2_CGS )
+                                      * SQR( 0.25 * SrcTerms.Lightbulb_Tnue );
+            const double rate_cooling = 1.399e20 * CUBE(  SQR( 0.5 * Temp_MeV )  );
+
+            const double tau       = 1.0e-11 * Dens_Code * UNIT_D;
+            const double rate_CGS  = ( rate_heating - rate_cooling ) * ( Xn + Xp ) * exp( -tau );
+
+            const double dEint_Code  = rate_CGS * UNIT_T * sEint2Code * Dens_Code;  // dEint per UNIT_T
+            const double _Eint_Ratio = fabs( dEint_Code / Eint_Code );
+
+//          compare the inverse of ratio to avoid zero division, and store the maximum value
+            if ( _Eint_Ratio > OMP_dt_LB_Inv[TID][0] )   OMP_dt_LB_Inv[TID][0] = _Eint_Ratio;
+
+         }}} // i,j,k
+      } // for (int PID=0; PID<amr->NPatchComma[lv][1]; PID++)
+   } // OpenMP parallel region
+
+
+// find the maximum over all OpenMP threads
+   for (int TID=0; TID<NT; TID++)
+   {
+      if ( OMP_dt_LB_Inv[TID][0] > dt_LB_Inv )   dt_LB_Inv = OMP_dt_LB_Inv[TID][0];
+   }
+
+// free per-thread arrays
+   Aux_DeallocateArray2D( OMP_dt_LB_Inv );
+
+
+// collect data from all ranks
+#  ifndef SERIAL
+   {
+      double dt_LB_Inv_All[MPI_NRank];
+
+      MPI_Allgather( &dt_LB_Inv, 1, MPI_DOUBLE, dt_LB_Inv_All, 1, MPI_DOUBLE, MPI_COMM_WORLD );
+
+      for (int i=0; i<MPI_NRank; i++)
+      {
+         if (  ( dt_LB_Inv_All[i] > 0.0 )  &&  ( 1.0 / dt_LB_Inv_All[i] < dt_LB )  )
+            dt_LB = 1.0 / dt_LB_Inv_All[i];
+      }
+   }
```

Comment:
- Replace `MPI_Allgather()` with `MPI_Allreduce()`. Also fix bugs if serial runs are performed.
- I check `dt_LB_Inv_All > 0.0` to ensure zero division will not occur. However, it seems this may not happen so I removed it.

Reply:
- For **in-place**`MPI_Allreduce()`, you should use `MPI_IN_PLACE` in the send buffer
- If you instead use **out-of-place** operation, I don't think you need to add `#  ifndef SERIA` (see `include/Serial.h`).